analysis/pfe_methods/unicode_range_pfe_method.py

```diff
@@ -0,0 +1,129 @@
+"""Implementation of PFE that breaks a font into a set of language based subsets.
+
+A single font gets broken into one or more subsets (typically one subset per script).
+The client then chooses to download (via unicode range) the set of subsets which
+contain characters on in use on a given page. For a single session downloaded subsets
+are cached and re-used.
+
+Cut subsets are woff2 encoded to minimize their size.
+
+Subset definitions are provided in //analysis/pfe_methods/unicode_range_data.
+These definitions are based on what Google Fonts uses for their production
+font serving.
+"""
+
+import io
+import os
+
+from analysis import request_graph
+from analysis.pfe_methods.unicode_range_data import slicing_strategy_loader
+from fontTools import subset
+from woff2_py import woff2
+
+# Cache of which slicing strategy to use per font. Keyed by font name.
+FONT_SLICING_STRATEGY_CACHE = dict()
+
+# Cache of cut and woff2 encoded subset sizes. Keyed by:
+# <font name>:<slicing_strategy>:<subset index>
+SUBSET_SIZE_CACHE = dict()
+
+
+def name():
+  return "UnicodeRange"
```

Comment:
Consider specifying this is Google Fonts Unicode Range since other versions could reasonably exist